packages/pigeon/lib/swift_generator.dart

```diff
@@ -0,0 +1,645 @@
+    'Uint8List': '[UInt8]',
+    'Int32List': '[Int32]',
+    'Int64List': '[Int64]',
+    'Float32List': '[Float32]',
+    'Float64List': '[Float64]',
+    'Object': 'Any',
+  };
+  if (swiftTypeForDartTypeMap.containsKey(type.baseName)) {
+    return swiftTypeForDartTypeMap[type.baseName];
+  } else if (type.baseName == 'List' || type.baseName == 'Map') {
+    return _swiftTypeForBuiltinGenericDartType(prefix, type);
+  } else {
+    return null;
+  }
+}
+
+String _swiftTypeForDartType(String? prefix, TypeDeclaration type) {
+  return _swiftTypeForBuiltinDartType(prefix, type) ??
+      _className(prefix, type.baseName);
+}
+
+String _nullsafeSwiftTypeForDartType(String? prefix, TypeDeclaration type) {
+  final String nullSafe = type.isNullable ? '?' : '';
+  return '${_swiftTypeForDartType(prefix, type)}$nullSafe';
+}
+
+/// Calculates the Swift class name, possibly prefixed.
+String _className(String? prefix, String className) {
+  if (prefix != null) {
+    return '$prefix$className';
+  } else {
+    return className;
+  }
+}
+
+/// Generates the ".swift" file for the AST represented by [root] to [sink] with the
+/// provided [options].
+void generateSwift(SwiftOptions options, Root root, StringSink sink) {
+  final Set<String> rootClassNameSet =
+      root.classes.map((Class x) => x.name).toSet();
+  final Set<String> rootEnumNameSet =
+      root.enums.map((Enum x) => x.name).toSet();
+  final Indent indent = Indent(sink);
+
+  HostDatatype _getHostDatatype(NamedType field) {
+    return getHostDatatype(field, root.classes, root.enums,
+        (NamedType x) => _swiftTypeForBuiltinDartType(options.prefix, x.type));
+  }
+
+  void writeHeader() {
+    if (options.copyrightHeader != null) {
+      addLines(indent, options.copyrightHeader!, linePrefix: '// ');
+    }
+    indent.writeln('// $generatedCodeWarning');
+    indent.writeln('// $seeAlsoWarning');
+  }
+
+  void writeImports() {
+    indent.writeln('import Foundation');
+    indent.writeln('import Flutter');
+  }
+
+  void writeEnum(Enum anEnum) {
+    indent.write('enum ${_className(options.prefix, anEnum.name)}: Int ');
+    indent.scoped('{', '}', () {
+      // We use explicit indexing here as use of the ordinal() method is
+      // discouraged. The toMap and fromMap API matches class API to allow
+      // the same code to work with enums and classes, but this
+      // can also be done directly in the host and flutter APIs.
+      int index = 0;
+      for (final String member in anEnum.members) {
+        indent.writeln('case $member = $index');
+        index++;
+      }
+    });
+  }
+
+  void writeDataClass(Class klass) {
+    void writeField(NamedType field) {
+      indent.write(
+          'var ${field.name}: ${_nullsafeSwiftTypeForDartType(options.prefix, field.type)}');
+      final String defaultNil = field.type.isNullable ? ' = nil' : '';
+      indent.addln(defaultNil);
+    }
+
+    void writeToMap() {
+      indent.write('func toMap() -> [String: Any?] ');
```

Comment:
We have to make sure that this isn't public.  We want the ability to change the wire format of pigeon so we can't let people rely on this.  I think in swift's case we can just make these methods `internal`.

Reply:
As this method as `.fromMap` are used in tests to validate the conversion, so do in ObjC.
By default, it is internal in Swift. Thus, any code outside the module couldn't access that.